tools/mo/openvino/tools/mo/ops/convolution.py

```diff
@@ -115,8 +115,16 @@ def infer(node: Node):
                     'group') and node.has_valid('kernel_spatial')):
                 log.error('Cannot reshape kernel due to not all required attrs was set to {} node'.format(node.id))
                 return
+
+            #  in shape_array([..., num_in_channels / node.group, ...])
+            #  num_in_channels / node.group must be a scalar
+            if len(np.array(node.channel_dims).shape) == 0:  # if channel_dims is scalar
+                num_in_channels = input_shape[node.channel_dims]
+            else:  # if channel_dims is an array
+                num_in_channels = input_shape[node.channel_dims][0]
```

Comment:
Agree. It looks ugly. I implemented a more clean solution. I replaced these lines with 

```
# since item() unmasks values, result should be masked 
backnum_in_channels = shape_array(input_shape[node.channel_dims].item())
```

The reason this is needed is that when `item()` is called it unmasks dynamic dimensions: if `input_shape = [1, dyn, dyn, dyn]` then `input_shape[node.channel_dims].item()` will be equal to `dynamic_dimension_value == -1000000007` instead of `dynamic_dimension == masked`
https://github.com/openvinotoolkit/openvino/blob/master/tools/mo/openvino/tools/mo/ops/convolution.py#L119

To avoid that I referred to elements with `[0]` instead of `item()`. But now solution with `shape_array(...)` is much more clean.